UOP1_Project/Assets/Scripts/Sound/SoundEmitter.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Audio;
+
+[RequireComponent (typeof(AudioSource))]
+public class SoundEmitter : MonoBehaviour
+{
+	private AudioSource _audioSource;
+
+	private void Awake()
+	{
+		DontDestroyOnLoad(this);
+		_audioSource = this.GetComponent<AudioSource>();
+	}
+
+	// Start is called before the first frame update
+	void Start()
+    {
+		_audioSource.playOnAwake = false;
+    }
```

Comment:
For now its just a simple wrapper but to add functionality like moving sounds we need monobehaviour, since afaik there is no trigger for when a sound is no longer playing, so we would need to use a coroutine to trigger events when its finished playing with the clip.length such as unparenting from the moving transform that is emitting the sound.

Reply:
Makes sense! Although that made me think about making a dispatcher... I might work on that 🙂